<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>801</width>
    <height>599</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="sliderLayout">
      <item>
       <widget class="QwtSlider" name="timeSlider">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="scalePosition">
         <enum>QwtSlider::LeadingScale</enum>
        </property>
        <property name="trough">
         <bool>false</bool>
        </property>
        <property name="groove">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="maxvalEdit">
        <property name="toolTip">
         <string>Change maximum time</string>
        </property>
        <property name="icon">
         <iconset resource="mainwindow.qrc">
          <normaloff>:/images/standardbutton-yes-32.png</normaloff>:/images/standardbutton-yes-32.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Visual::HistoryScreen" name="historyScreen"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>801</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="playbackToolBar">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="windowTitle">
    <string>Playback toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionPlay"/>
   <addaction name="actionStop"/>
   <addaction name="actionHome"/>
   <addaction name="actionEnd"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="fileToolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="windowTitle">
    <string>File toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>RightToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLoadMap"/>
   <addaction name="actionOpenFile"/>
   <addaction name="actionSaveFile"/>
  </widget>
  <action name="actionPlay">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/media-play-32.png</normaloff>
     <normalon>:/images/media-pause-32.png</normalon>:/images/media-play-32.png</iconset>
   </property>
   <property name="text">
    <string>Play/Pause</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/media-stop-32.png</normaloff>:/images/media-stop-32.png</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
  <action name="actionHome">
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/media-skip-backward-32.png</normaloff>:/images/media-skip-backward-32.png</iconset>
   </property>
   <property name="text">
    <string>To the beginning</string>
   </property>
   <property name="shortcut">
    <string>Home</string>
   </property>
  </action>
  <action name="actionEnd">
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/media-skip-forward-32.png</normaloff>:/images/media-skip-forward-32.png</iconset>
   </property>
   <property name="text">
    <string>To the end</string>
   </property>
   <property name="shortcut">
    <string>End</string>
   </property>
  </action>
  <action name="actionLoadMap">
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/file-load-map-32.png</normaloff>:/images/file-load-map-32.png</iconset>
   </property>
   <property name="text">
    <string>Load Velocity map</string>
   </property>
   <property name="iconText">
    <string>Load Velocity map </string>
   </property>
   <property name="toolTip">
    <string>Load velocity map from a text file (Ctrl+M)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="actionOpenFile">
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/file-open-32.png</normaloff>:/images/file-open-32.png</iconset>
   </property>
   <property name="text">
    <string>Open File</string>
   </property>
   <property name="toolTip">
    <string>Open File (Ctrl+O)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSaveFile">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="mainwindow.qrc">
     <normaloff>:/images/file-save-32.png</normaloff>:/images/file-save-32.png</iconset>
   </property>
   <property name="text">
    <string>Save File</string>
   </property>
   <property name="toolTip">
    <string>Save File (Ctrl+S)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QwtSlider</class>
   <extends>QWidget</extends>
   <header>qwt_slider.h</header>
  </customwidget>
  <customwidget>
   <class>Visual::HistoryScreen</class>
   <extends>QGraphicsView</extends>
   <header>historyscreen.h</header>
   <slots>
    <signal>at(qreal)</signal>
    <signal>paused()</signal>
    <signal>maxtimeChanged(qreal)</signal>
    <slot>seek(qreal)</slot>
    <slot>playpause()</slot>
    <slot>stop()</slot>
    <slot>seek_home()</slot>
    <slot>seek_end()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="mainwindow.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionPlay</sender>
   <signal>triggered(bool)</signal>
   <receiver>historyScreen</receiver>
   <slot>playpause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStop</sender>
   <signal>triggered()</signal>
   <receiver>historyScreen</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>historyScreen</sender>
   <signal>at(qreal)</signal>
   <receiver>MainWindow</receiver>
   <slot>moveSlider(qreal)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>336</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStop</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>untogglePlay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>historyScreen</sender>
   <signal>paused()</signal>
   <receiver>MainWindow</receiver>
   <slot>untogglePlay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>336</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionHome</sender>
   <signal>triggered()</signal>
   <receiver>historyScreen</receiver>
   <slot>seek_home()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEnd</sender>
   <signal>triggered()</signal>
   <receiver>historyScreen</receiver>
   <slot>seek_end()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>historyScreen</sender>
   <signal>maxtimeChanged(qreal)</signal>
   <receiver>MainWindow</receiver>
   <slot>setLimits(qreal)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>205</y>
    </hint>
    <hint type="destinationlabel">
     <x>103</x>
     <y>88</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>timeSlider</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>seekTime(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>382</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>loaded(bool)</signal>
   <receiver>actionSaveFile</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>loaded(bool)</signal>
   <receiver>playbackToolBar</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>400</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxvalEdit</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>requestMaxtime()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>735</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>763</x>
     <y>121</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>loaded(bool)</signal>
  <slot>seekTime(double)</slot>
  <slot>moveSlider(qreal)</slot>
  <slot>untogglePlay()</slot>
  <slot>setLimits(qreal)</slot>
  <slot>requestMaxtime()</slot>
 </slots>
</ui>
